openfga/v1/authzmodel.proto

```diff
@@ -59,6 +61,26 @@ message TypeDefinition {
       example: "{\"relations\":{\"reader\":{\"union\":{\"child\":[{\"this\":{}},{\"computedUserset\":{\"object\":\"\",\"relation\":\"writer\"}}]}},\"writer\":{\"this\":{}}}}"
     }
   ];
+
+  // A map whose keys are the name of the relation and whose value is the TypeInformation for that
+  // relation.
+  map<string, TypeInformation> type_info = 3;
+}
+
+message TypeInformation {
+  repeated RelationReference assignable_relations = 1;
```

Comment:
Will this result in the below json?
```
"type_info": {
  "editor": {
    "assignable_relations": {
      ...
    }
  }
}
```

Do we need the assignable relations indirection when it comes to types?

To allow for other options, maybe instead of `type_info`, we can call it `metadata`?
```
"metadata": {
  "editor": {
    "comment": "...",
    "assignable_relations": {
      ...
    }
  }
}
```

Reply:
I think the JSON will look like:
```
{
  "type": "document",
  "relations": {
    "admin": ...
  },
  "type_info": {
    "admin": {
      "assignable_relations": [
        {
          "type": "user",
        },
        {
          "type": "group",
          "relation": "member"
        }
      ]
    }
  }
}
```
I don't mind calling it something else if you would like. `metadata`?

Reply:
If we have assignable relations field, I think something generic like that would make more sense and would allow us to extend to other options related to relations

Reply:
I like `metadata`